src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionBuilder.cs

```diff
@@ -244,6 +246,14 @@ private List<(string ResourceType, SearchParameterInfo SearchParameter)> Validat
                                 componentIndex);
                             continue;
                         }
+
+                        resolvedComponents.Add(componentSearchParameter);
+                    }
+
+                    if (resolvedComponents.Count > 0)
+                    {
+                        // this is not actually creating the search parameter, it will return the existing one.
+                        CreateSearchParameterInfo(searchParameter).ResolvedComponents = resolvedComponents;
```

Comment:
>CreateSearchParameterInfo [](start = 24, length = 25)

nit: this reads like some kind of black magic. Can we rename to `GetOrCreateSearchParameterInfo`

Reply:
or just do `_urlDictionary.GetValue(searchParameter.Uri).ResolvedComponents = resolvedComponents`


---
In reply to: [538846387](https://github.com/microsoft/fhir-server/pull/1497#discussion_r538846387) [](ancestors = 538846387)

Reply:
Renamed.